### ISSUE TITLE
OmniSharp stop loading after first project in workspace

### ISSUE THREAD

**Issue by alaconda** [OmniSharp/omnisharp-vscode#3988]
## Environment data
`dotnet --info` output:
.NET Core SDK (reflecting any global.json):
 Version:   3.1.401
 Commit:    39d17847db

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  18.04
 OS Platform: Linux
 RID:         ubuntu.18.04-x64
 Base Path:   /usr/share/dotnet/sdk/3.1.401/

Host (useful for support):
  Version: 3.1.7
  Commit:  fcfdef8d6b

.NET Core SDKs installed:
  2.1.809 [/usr/share/dotnet/sdk]
  3.0.103 [/usr/share/dotnet/sdk]
  3.1.401 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.21 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.21 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.3 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.7 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.21 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.3 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.7 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

VS Code version: 1.48.0
C# Extension version: 1.23.0

## OmniSharp log
Starting OmniSharp server at 2020-08-16, 2:04:46 p.m.
    Target: /home/alaconda/Workspaces/HealthKnex/hknex

OmniSharp server started.
    Path: /home/alaconda/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/run
    PID: 9433

[info]: OmniSharp.Stdio.Host
        Starting OmniSharp on ubuntu 18.4 (x64)
[info]: OmniSharp.Services.DotNetCliService
        DotNetPath set to dotnet
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Located 1 MSBuild instance(s)
            1: StandAlone 16.8.0 - "/home/alaconda/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin"
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        MSBUILD_EXE_PATH environment variable set to '/home/alaconda/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin/MSBuild.dll'
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Registered MSBuild instance: StandAlone 16.8.0 - "/home/alaconda/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin"
            CscToolExe = csc.exe
            MSBuildToolsPath = /home/alaconda/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin
            CscToolPath = /home/alaconda/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin/Roslyn
            BypassFrameworkInstallChecks = true
            MSBuildExtensionsPath = /home/alaconda/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild
[info]: OmniSharp.Cake.CakeProjectSystem
        Detecting Cake files in '/home/alaconda/Workspaces/HealthKnex/hknex'.
[info]: OmniSharp.Cake.CakeProjectSystem
        Could not find any Cake files
[info]: OmniSharp.MSBuild.ProjectSystem
        No solution files found in '/home/alaconda/Workspaces/HealthKnex/hknex'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for '/home/alaconda/Workspaces/HealthKnex/hknex/hknex.csproj'
[info]: OmniSharp.Script.ScriptProjectSystem
        Detecting CSX files in '/home/alaconda/Workspaces/HealthKnex/hknex'.
[info]: OmniSharp.Script.ScriptProjectSystem
        Could not find any CSX files
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.CSharpFormattingWorkspaceOptionsProvider, Order: 0
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.RenameWorkspaceOptionsProvider, Order: 100
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.ImplementTypeWorkspaceOptionsProvider, Order: 110
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.BlockStructureWorkspaceOptionsProvider, Order: 140
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: /home/alaconda/Workspaces/HealthKnex/hknex/hknex.csproj
[info]: OmniSharp.WorkspaceInitializer
        Configuration finished.
[info]: OmniSharp.Stdio.Host
        Omnisharp server running using Stdio at location '/home/alaconda/Workspaces/HealthKnex/hknex' on host 9088.
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file '/home/alaconda/Workspaces/HealthKnex/hknex/hknex.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Adding project '/home/alaconda/Workspaces/HealthKnex/hknex/hknex.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: hknex

## Steps to reproduce
Open any workspace.

## Expected  behavior
OmniSharp to load and update all projects in workspace.

## Actual behavior
OmniSharp seems to stop loading after the first project in the workspace.

This might be related to #3987 and #3983 

#3983 In my case, the intellisense seems is working for the first project, but not the subsequent projects.
#3987 In my workspace, the second project has a project reference to first project, and if I load second project independently it will show the same warnings of "cannot be imported again" in #3987.  Not sure if this is the root cause of OmniSharp not being able to load after first project.

It should be noted that if I load each project independently, OmniSharp will show the "cannot be imported again" warnings on all projects that have project references, but intellisense will work.

**Comment by alaconda** [comment]
**Update:** I was able to get OmniSharp to load the rest of the projects in workspace by setting `omnisharp.enableMsBuildLoadProjectsOnDemand = true`.

However, I am still getting the warning messages of `cannot be imported again` for each of the projects that has a project reference.

![image](https://user-images.githubusercontent.com/40466226/90343088-5b406480-dfdb-11ea-882a-39658c86a9aa.png)

> /home/alaconda/Workspaces/HealthKnex/hknex-data/hknex-data.csproj
> /home/alaconda/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin/Microsoft.Xaml.targets(19,4): Error: "/home/alaconda/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin/Microsoft.Xaml.targets" cannot be imported again. It was already imported at "/home/alaconda/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin/Microsoft.Common.CurrentVersion.targets (6156,3)". This is most likely a build authoring error. This subsequent import will be ignored. [/home/alaconda/Workspaces/HealthKnex/hknex/hknex.csproj]
> /home/alaconda/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin/Microsoft.WorkflowBuildExtensions.targets(25,4): Error: "/home/alaconda/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin/Microsoft.WorkflowBuildExtensions.targets" cannot be imported again. It was already imported at "/home/alaconda/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin/Microsoft.Common.CurrentVersion.targets (6160,3)". This is most likely a build authoring error. This subsequent import will be ignored. [/home/alaconda/Workspaces/HealthKnex/hknex/hknex.csproj]
> /home/alaconda/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin/Microsoft.ServiceModel.targets(12,4): Error: "/home/alaconda/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin/Microsoft.ServiceModel.targets" cannot be imported again. It was already imported at "/home/alaconda/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin/Microsoft.CSharp.CurrentVersion.targets (343,5)". This is most likely a build authoring error. This subsequent import will be ignored. [/home/alaconda/Workspaces/HealthKnex/hknex/hknex.csproj]

**Comment by broder-peters-k24** [comment]
I'm also getting the `cannot be imported again` errors/warning. Everything else seems to work fine at the end but it still bugs me to always get the notification and I'm not really sure what to do with the error information.

```
"/home/broderpeters/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin/Microsoft.Xaml.targets" cannot be imported again. It was already imported at "/home/broderpeters/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin/Microsoft.Common.CurrentVersion.targets (6156,3)". This is most likely a build authoring error. This subsequent import will be ignored. 
```

**Comment by broder-peters-k24** [comment]
> I'm also getting the `cannot be imported again` errors/warning. Everything else seems to work fine at the end but it still bugs me to always get the notification and I'm not really sure what to do with the error information.
> 
> ```
> "/home/broderpeters/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin/Microsoft.Xaml.targets" cannot be imported again. It was already imported at "/home/broderpeters/.vscode/extensions/ms-dotnettools.csharp-1.23.0/.omnisharp/1.36.1/omnisharp/.msbuild/Current/Bin/Microsoft.Common.CurrentVersion.targets (6156,3)". This is most likely a build authoring error. This subsequent import will be ignored. 
> ```

Just found out that there is already https://github.com/OmniSharp/omnisharp-vscode/issues/3987 for this issue.

**Comment by filipw** [comment]
This has been resolved in 1.23.1
